pkg/inject/sidecar.go

```diff
@@ -192,9 +192,9 @@ func container(jaeger *v1.Jaeger, dep *appsv1.Deployment) corev1.Container {
 
 	commonSpec := util.Merge([]v1.JaegerCommonSpec{jaeger.Spec.Agent.JaegerCommonSpec, jaeger.Spec.JaegerCommonSpec})
 
-	// Use a different common spec for volumes and mounts.
+	// Use only the agent common spec for volumes and mounts.
 	// We don't want to mount all Jaeger internal volumes into user's deployments
-	volumesAndMountsSpec := &v1.JaegerCommonSpec{}
+	volumesAndMountsSpec := &jaeger.Spec.Agent.JaegerCommonSpec
 	otelConf, err := jaeger.Spec.Agent.Config.GetMap()
```

Comment:
This would require users to always specify the volume under the `agent` node in order to be consumed here, right? Meaning, if I specified a volume and mount with the TLS at the root of the CR, it will be applied to every component *except* to the deployments where the sidecar has been injected.

I like this approach, perhaps @objectiser and/or @pavolloffay would also want to give their opinion.

Reply:
Yes, only the volumes under `spec.agent` would be added to the deployment, not those under just `spec`.

This was done to satisfy the comment in the code: 
`// We don't want to mount all Jaeger internal volumes into user's deployments`

So all the common spec volumes aren't added to the deployment still, but if someone specifically _does_ want a volume & volumeMount added to the sidecar's deployment & sidecar container respectively, they can specify it under the `agent` node.